frontend/www/js/omegaup/components/contest/AddProblem.test.ts

```diff
@@ -61,8 +62,62 @@ describe('AddProblem.vue', () => {
       },
     });
 
-    expect(wrapper.find('td[data-remove-problem] button').text()).toContain(
-      '🚫',
+    expect(wrapper.find('button[data-remove-problem-disabaled]')).toBeTruthy();
+  });
+
+  it('Should update a problem in the list', async () => {
+    const wrapper = shallowMount(contest_AddProblem, {
+      propsData: {
+        contestAlias: 'testContestAlias',
+        initialPoints: 100,
+        initialProblems: [
+          { ...problem, has_submissions: false },
+          { ...problem, alias: 'problem_2', letter: 'B', title: 'Problem 2' },
+        ],
+      },
+    });
+
+    expect(wrapper.find('form').text()).not.toContain(
+      T.contestAddproblemChooseVersion,
+    );
+    expect(wrapper.find('form').text()).not.toContain(T.wordsPoints);
+    expect(wrapper.find('form').text()).not.toContain(
+      T.contestAddproblemProblemOrder,
+    );
+
+    expect(wrapper.find('button[data-update-problem]')).toBeTruthy();
+    await wrapper.find('button[data-update-problem]').trigger('click');
+
+    expect(wrapper.find('form').text()).toContain(
+      T.contestAddproblemChooseVersion,
     );
+    expect(wrapper.find('form').text()).toContain(T.wordsPoints);
+    expect(wrapper.find('form').text()).toContain(
+      T.contestAddproblemProblemOrder,
+    );
+    expect(wrapper.find('button.add-problem').text()).toBe(
+      T.wordsUpdateProblem,
+    );
+
+    const updatedProblem = {
+      order: '2',
+      points: '98',
+    };
+    wrapper.setData(updatedProblem);
+
+    await wrapper.find('button.add-problem').trigger('click');
+    expect(wrapper.emitted('add-problem')).toBeDefined();
+    expect(wrapper.emitted('add-problem')).toEqual([
+      [
+        {
+          isUpdate: true,
+          problem: {
+            ...updatedProblem,
+            alias: 'problem',
+            commit: undefined,
```

Comment:
```suggestion
            commit: '54318b04024976471c4cac1d2efeb81021d9396b',
```

si se manda `undefined`, estaríamos accidentalmente sobreescribiendo la versión del problema a la última accidentalmente, aunque la persona que está editando el concurso no quiera!

se pueden agregar otras dos pruebas, uno donde se haga click explícitamente en el radio donde la persona quiere usar la última versión, y otra donde se elija otra versión?

Reply:
Estoy tratando de entender que es lo que hace actualmente, en el `$emit` se manda lo siguiente para el valor de `commit`:
```javascript
!this.useLatestVersion && this.selectedRevision ? this.selectedRevision.commit  : undefined,
```
que quiere decir que si no está seleccionada la opción de usar la última versión y `selectedRevision` trae algo (lo que sea, cosa que siempre será verdadero porque esto se inicializa con el valor de `commitEmpty`) debería obtener el commit de la revisión seleccionada. En el caso contrario lo mandará como `undefined`, cosa que no debería de afectar porque en el controlador se valida que si  `commit` es vacío o no existe, use el commit con el que está publicado el problema:
```php
        if (is_null($commit)) {
            $masterCommit = (new \OmegaUp\ProblemArtifacts(
                strval($problem->alias),
                'published'
            ))->commit();
        }
```
, así que en teoría no debería haber problema si mandamos ese `undefined` en el `commit`.

De cualquier forma haré el cambio, ya que no tiene sentido mandar eso si podemos acceder al valor, y esto haría más legible el código.

Me dices si con la prueba que agregué es suficiente o agrego las dos que sugieres (faltaría la de picar el botón de Última versión', y es la opción por defecto).

Reply:
sí hay problema si se manda `undefined`, porque el caso que me preocupa es cuando se publica una nueva versión, alguien _no_ quiere actualizar a esa versión, y al actualizar el problema se rejuecea todo sin querer.

Reply:
Tienes razón. Que bueno que me dijiste, ya con este cambio vamos a evitar esa posible falla.